lib/WorseReflection/Core/Reflection/Collection/ReflectionClassLikeCollection.php

```diff
@@ -55,6 +56,12 @@ public static function fromNode(ServiceLocator $serviceLocator, TextDocument $so
 
             if ($child instanceof ClassDeclaration) {
                 $items[(string) $child->getNamespacedName()] = new ReflectionClass($serviceLocator, $source, $child, $visited);
+                continue;
+            }
+
+            if ($child instanceof ObjectCreationExpression) {
+                // TODO: come up with a good way to generate a class name
```

Comment:
PHPStan does it like this: https://github.com/dantleech/phpstan-src/blob/f7b89d47cfa1fa636fb571b4315841a9af0e976f/src/Broker/AnonymousClassNameHelper.php#L22-L25

but I don't think this is the correct place to do it.

We can create the reflection class in the resolver?

Reply:
Yeah, technically using the line number is not 100% garanteed to be unique but who has two anon classes on the same line. :D

But what you mean with "creating the reflection class in the resolver"? I wanted to try to avoid making a `ReflectionAnonymousClass` class as anon classes should be the same except for their weird name.

Reply:
I don't think it needs refactoring and would also fix the issue Thomas raised below, but would need to show an example.